DllImportGenerator/DllImportGenerator/Marshalling/MarshallingGenerator.cs

```diff
@@ -263,5 +275,64 @@ private static IMarshallingGenerator CreateStringMarshaller(TypePositionInfo inf
 
             throw new MarshallingNotSupportedException(info, context);
         }
+
+        
+        private static ExpressionSyntax GetNumElementsExpressionFromMarshallingInfo(TypePositionInfo info, StubCodeContext context)
+        {
+            ExpressionSyntax numElementsExpression;
+            if (info.MarshallingAttributeInfo is MarshalAsInfo { ArraySizeParamIndex: short index, ArraySizeConst: int constSize })
+            {
+                LiteralExpressionSyntax? constSizeExpression = constSize != MarshalAsInfo.UnspecifiedData
+                    ? LiteralExpression(SyntaxKind.NumericLiteralExpression, Literal(constSize))
+                    : null;
+                ExpressionSyntax? sizeParamIndexExpression = null;
+                if (index != MarshalAsInfo.UnspecifiedData)
+                {
+                    TypePositionInfo? paramIndexInfo = context.GetTypePositionInfoForManagedIndex(index);
+                    if (paramIndexInfo is null)
+                    {
+                        throw new MarshallingNotSupportedException(info, context);
```

Comment:
Can we make this exception more precise here. I think this really means the caller didn't provide an index correct? If this will be a specific message then okay, otherwise we should provide details.